src/main/java/com/enderio/base/common/blockentity/IAutoEquiping.java

```diff
@@ -0,0 +1,14 @@
+package com.enderio.base.common.blockentity;
+
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.context.UseOnContext;
+
+/**
+ * An interface that block entities may implement in order to support item right-click behaviour. Used for auto-equipping capacitors on supported machines.
+ */
+public interface IAutoEquiping {
```

Comment:
Maybe go for sometime like IWorldInsert or something rather than equipping? Or IWorldInstall (but install is more a word just used for capacitors, so maybe insert is just better here)

Reply:
"Install" probably is a better word given that interactions are limited to capacitors. Upgrade system can use this in future and "install" also works with it.
I am not sure if "world" should be used. How about "IMachineInstall"

Reply:
Im happy with machine install :)